test/tests/VehicleTest.cpp

```diff
@@ -0,0 +1,1218 @@
+                {
+                    if (RideTypeDescriptors[ride->type].StartTrackPiece != FLAT_TRACK_ELEM_1_X_4_A)
+                    {
+                        if (ride->type == RIDE_TYPE_ENTERPRISE)
+                        {
+                            direction += 5;
+                        }
+                        else
+                        {
+                            direction = 4;
+                        }
+                    }
+                }
+            }
+        }
+
+        chosenLoc += CoordsXYZ{ word_9A2A60[direction], RideTypeDescriptors[ride->type].Heights.VehicleZOffset };
+
+        vehicle->current_station = trackElement->GetStationIndex();
+
+        vehicle->MoveTo(chosenLoc);
+        vehicle->SetTrackType(trackElement->GetTrackType());
+        vehicle->SetTrackDirection(vehicle->sprite_direction >> 3);
+        vehicle->track_progress = 31;
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_MINI_GOLF)
+        {
+            vehicle->track_progress = 15;
+        }
+        vehicle->update_flags = VEHICLE_UPDATE_FLAG_COLLISION_DISABLED;
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_HAS_INVERTED_SPRITE_SET)
+        {
+            if (trackElement->IsInverted())
+            {
+                vehicle->SetUpdateFlag(VEHICLE_UPDATE_FLAG_USE_INVERTED_SPRITES);
+            }
+        }
+        vehicle->SetState(Vehicle::Status::MovingToEndOfStation);
+    }
+
+    // loc_6DDD5E:
+    vehicle->num_peeps = 0;
+    vehicle->next_free_seat = 0;
+    vehicle->BoatLocation.setNull();
+    return vehicle;
+}
+
+rct_ride_entry* Ride::GetRideEntry() const
+{
+    return get_ride_entry(subtype);
+}
+
+TEST(Vehicle, VehicleCreateSimple)
+{
+    reset_sprite_list();
+    Ride& r = _rideList[0];
+    r = Ride{};
+    r.id = 0;
+    r.type = RIDE_TYPE_MONORAIL;
+    int32_t rd{};
+    TrackElement te{};
+    vehicle_create_car(0, 0, 0, 0, {}, &rd, &te);
+}
+
+static std::vector<std::pair<std::string, int>> _goto_callsites;
+
+// Silence the -Werror=missing-declarations error
+void hit_goto_callsite(std::string target, int callsite_id);
+void hit_goto_callsite(std::string target, int callsite_id)
+{
+    _goto_callsites.push_back(std::make_pair(target, callsite_id));
+}
+
+TEST(Vehicle, GotoCallsite)
+{
+    _goto_callsites.clear();
+    reset_sprite_list();
+    Ride& r = _rideList[0];
+    r = Ride{};
+    r.id = 0;
+    r.type = RIDE_TYPE_MONORAIL;
+    r.mode = RideMode::Race;
+    r.lifecycle_flags = RIDE_LIFECYCLE_PASS_STATION_NO_STOPPING;
+    int32_t rd{};
+    TrackElement te{};
+    Vehicle* v = vehicle_create_car(0, 0, 0, 0, {}, &rd, &te);
+    v->status = Vehicle::Status::Arriving;
+    v->Update();
+    EXPECT_EQ(_goto_callsites.size(), 1);
+    std::vector<std::pair<std::string, int>> expected_results;
+    expected_results.push_back(std::make_pair("loc_6D8E36", 1));
+    EXPECT_EQ(_goto_callsites, expected_results);
```

Comment:
I find this the most exciting part of this whole PR. I've added a temporary call to a function `hit_goto_callsite` provided just above this test, which I can use to ensure all the callsites I'm interested in are hit. Once I'm happy with the quality of the tests, the next step would be gathering checksums of all affected objects after the test is done and removing the calls to `hit_goto_callsite`